app/routes.py

```diff
@@ -644,8 +618,14 @@ def update_exchange():
     if not current_user.is_authenticated:
         return jsonify(target="login?next="+data['state'])
     if data["task"] == "verify":
-        exchanges = db.session.query(Exchange)
-        exchanges = exchanges.filter_by(exchange_id=int(data["exchange_id"]))
+        exchanges = db.session.query(Exchange).filter_by(exchange_id=int(data["exchange_id"]))
+        eid = -1
+        try:
+            eid = int(data['e_id'])
+        except:
+            pass
+        if eid >= 0:
+            exchanges = exchanges.filter_by(id=eid) 
         for ex in exchanges:
             if ex.hospital1 == int(data["hospital_id"]):
```

Comment:
This is outside the scope of this PR, but we should be validating the hospital against the currently authenticated user, otherwise it's a huge security vulnerability.